tanjun/annotations.py

```diff
@@ -541,6 +542,101 @@ def set_config(self, config: _ArgConfig, /) -> None:
         config.is_greedy = True
 
 
+class _LengthMeta(abc.ABCMeta):
+    def __getitem__(cls, value: typing.Union[int, tuple[int, int]], /) -> type[str]:
+        if isinstance(value, int):
+            obj = Length(value)
+
+        else:
+            obj = Length(*value)
+
+        return typing.Annotated[Str, obj]
+
+
+class Length(_ConfigIdentifier, metaclass=_LengthMeta):
```

Comment:
Document how this might not play well with converters when implementing both slash and message commands as its applied before conversion with slash commands but after with message commands